django/db/models/functions.py

```diff
@@ -27,7 +27,7 @@ class ToNCLOB(Func):
             expressions = [
                 ToNCLOB(expression) for expression in self.get_source_expressions()]
             self.set_source_expressions(expressions)
```

Comment:
What this should do here is something like:

```
c = self.copy()
c.set_source_expressions(expressions)
return super(Coalesce, c).as_sql(compiler, connection, **kwargs)
```

The theme here is to stop mutating `self` within `as_sql` or `as_postgres/oracle` etc, which is the major point of adding `kwargs`.